searchlib/src/vespa/searchlib/attribute/attribute_operation.cpp

```diff
@@ -84,17 +84,19 @@ struct Set {
 
 template <typename T, typename OP>
 struct UpdateFast {
-    using A = search::SingleValueNumericAttribute<T>;
+    using A = SingleValueNumericAttribute<T>;
     using F = OP;
     A * attr;
     F op;
     typedef typename T::LoadedValueType ValueType;
-    UpdateFast(search::attribute::IAttributeVector &attr_in, typename F::V operand)
+    UpdateFast(IAttributeVector &attr_in, typename F::V operand)
         : attr(dynamic_cast<A *>(&attr_in)),
           op(operand)
     {}
     void operator()(uint32_t docid) { attr->set(docid, op(attr->getFast(docid))); }
-    bool valid() const { return (attr != nullptr); }
+    bool valid() const {
+        return (attr != nullptr) &&
+               (dynamic_cast<const AttributeVector &>(*attr).getConfig().isMutable()); }
```

Comment:
so, a SingleValueNumericAttribute<T> is not an AttributeVector, that seems kind of complicated...?